src/main/kotlin/org/rust/lang/core/lexer/RustEscapesLexer.kt

```diff
@@ -123,15 +112,30 @@ class RustEscapesLexer private constructor(val defaultToken: IElementType,
 
     companion object {
         fun forByteLiterals(): RustEscapesLexer =
-            RustEscapesLexer(RustTokenElementTypes.BYTE_LITERAL)
+            RustEscapesLexer(BYTE_LITERAL)
 
         fun forCharLiterals(): RustEscapesLexer =
-            RustEscapesLexer(RustTokenElementTypes.CHAR_LITERAL, unicode = true)
+            RustEscapesLexer(CHAR_LITERAL, unicode = true)
 
         fun forByteStringLiterals(): RustEscapesLexer =
-            RustEscapesLexer(RustTokenElementTypes.BYTE_STRING_LITERAL, eol = true)
+            RustEscapesLexer(BYTE_STRING_LITERAL, eol = true)
 
         fun forStringLiterals(): RustEscapesLexer =
-            RustEscapesLexer(RustTokenElementTypes.STRING_LITERAL, unicode = true, eol = true)
+            RustEscapesLexer(STRING_LITERAL, unicode = true, eol = true)
+
+        /**
+         * Create an instance of [RustEscapesLexer] suitable for given [IElementType].
+         *
+         * For the set of supported token types see [ESCAPABLE_LITERALS_TOKEN_SET].
+         *
+         * @throws IllegalArgumentException when given token type is unsupported
+         */
+        fun of(tokenType: IElementType): RustEscapesLexer = when (tokenType) {
+            BYTE_LITERAL        -> forByteLiterals()
+            CHAR_LITERAL        -> forCharLiterals()
+            BYTE_STRING_LITERAL -> forByteStringLiterals()
+            STRING_LITERAL      -> forStringLiterals()
+            else                -> throw IllegalArgumentException("unsupported literal type")
+        }
```

Comment:
My original intention was that the `of` is just a convenient shortcut for the `for` family, with a trade-off that it can raise an exception. `for`s are used when we want to handle only single literal type (currently they are used only in tests)